lib/rucio/web/ui/common/utils.py

```diff
@@ -152,3 +153,116 @@ def check_token(rendered_tpl):
+
+    # # try to get and check the rucio session token from cookie
+    session_token = cookies().get('x-rucio-auth-token')
+    validate_token = authentication.validate_auth_token(session_token)
+
+    # if token is valid, render the requested page.
+    if validate_token and not data:
+        token = session_token
+        js_token = __to_js('token', token)
+        js_account = __to_js('account', def_account)
+
+        return render.base(js_token, js_account, rucio_ui_version, policy, rendered_tpl)
+
+    else:
+        # if there is no session token or if invalid: get a new one.
+        # if user tries to access a page through URL without logging in, then redirect to login page.
+        if rendered_tpl:
+            return render.login()
+
+        # get all accounts for an identity. Needed for account switcher in UI.
+        accounts = identity.list_accounts_for_identity(data.username, 'userpass')
+        if len(accounts) == 0:
+            return render.problem('No accounts for the given identity.')
+
+        cookie_accounts = accounts
+        # try to set the default account to the user account, if not available take the first account.
+        def_account = accounts[0]
+        for account in accounts:
+            account_info = get_account_info(account)
+            if account_info.account_type == AccountType.USER:
+                def_account = account
+                break
+
+        selected_account = cookies().get('rucio-selected-account')
+        if (selected_account):
+            def_account = selected_account
+
+        try:
+            token = authentication.get_auth_token_user_pass(def_account,
+                                                            data.username,
+                                                            data.password.encode("ascii"),
+                                                            'webui',
+                                                            ctx.env.get('REMOTE_ADDR')).token
+
+        except:
+            return render.problem('Cannot get auth token')
+
+        attribs = list_account_attributes(def_account)
+        # write the token and account to javascript variables, that will be used in the HTML templates.
+        js_token = __to_js('token', token)
+        js_account = __to_js('account', def_account)
+
+    # # if there was no valid session token write the new token to a cookie.
+    if token:
+        setcookie('x-rucio-auth-token', value=token, path='/')
+        setcookie('rucio-auth-token-created-at', value=int(time()), path='/')
+
+    if cookie_accounts:
+        values = ""
+        for acc in cookie_accounts:
+            values += acc + " "
+        setcookie('rucio-available-accounts', value=values[:-1], path='/')
+
+    if attribs:
+        setcookie('rucio-account-attr', value=dumps(attribs), path='/')
+
+    return seeother('/')
+
+
+def authenticate(rendered_tpl):
+
+    auth_type = config_get('webui', 'auth_type')
```

Comment:
please put a try/except in case no 'webui' section or 'auth_type' is defined and use 'x509' as default.

Reply:
Okay Sir :+1: